README.md

```diff
@@ -236,7 +236,93 @@ TODO: Add notes about how to use the system.
 
 ## 🚀 Deployment <a name = "deployment"></a>
 
-TODO: Add additional notes about how to deploy this on a live system.
+### Prerequisites
+
+What things do you need to install the software, and how to install them.
+
+[git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+
+[docker compose](https://docs.docker.com/compose/install/)
+
+[make](https://www.gnu.org/software/make/)
+
+> **_NOTE:_**  make is optional, but you may find that having it greatly simplifies your life.
+
+### Installing
+
+Register a new OAuth application on the [42 intranet](https://profile.intra.42.fr/oauth/applications/new)
+
+Set the Redirect URIs
+
+```
+http://your-domain-or-ip-here/api/v1/auth/login
+http://localhost/api/v1/auth/login
+```
+
+Clone the GitHub repository
+
+```shell
+git clone https://github.com/42AGV/ft_transcendence.git
+```
+
+
+Change the current working directory to `ft_transcendence`
+
+```shell
+cd ft_transcendence
+```
+
+Either create a `.env` file in the root directory containing the following environment variables (replace the FORTYTWO_APP_ID and FORTYTWO_APP_SECRET with the ones from the OAuth application you create in the previous step):
+
+> **_NOTE:_**  You can generate UUIDs using the `uuidgen` command
+
+```
+FORTYTWO_APP_ID=your-42-app-id
+FORTYTWO_APP_SECRET=your-42-app-secret
+FORTYTWO_APP_CALLBACK_URL=http://your-domain-or-ip-here/api/v1/auth/login
+POSTGRES_USER=your-postgres-user
+POSTGRES_PASSWORD=your-postgres-password
+SESSION_SECRET=your-session-secret-uuid
+MEMCACHED_SECRET=your-memcached-secret-uuid
+```
+
+Or export them from the shell like:
+
+```shell
+export FORTYTWO_APP_ID=your-42-app-id
+export FORTYTWO_APP_SECRET=your-42-app-secret
+export FORTYTWO_APP_CALLBACK_URL=http://your-domain-or-ip-here/api/v1/auth/login
+export POSTGRES_USER=your-postgres-user
+export POSTGRES_PASSWORD=your-postgres-password
+export SESSION_SECRET=your-session-secret-uuid
+export MEMCACHED_SECRET=your-memcached-secret-uuid
+```
+
+Generate the generated files, that are not a part of the repository:
+
+```shell
+make gen
+```
+or
+```shell
+./scripts/generate-openapi.sh [options]
+```
```

Comment:
I'd just keep the make instructions in the readme. They are handier and nobody is going to use the scripts if you have a good makefile. Also, we want readme to be concise